### PR TITLE
BZ2029279: Removes the --index-filter-by-os flag

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.11+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2029279

Link to docs preview: http://file.rdu.redhat.com/sdudhgao/index-filter-by-os-flag/installing/disconnected_install/installing-mirroring-installation-images.html#olm-mirror-catalog-colocated_installing-mirroring-installation-images

Additional information:

### CODE REVIEW

File: modules/olm-mirroring-catalog-colocated.adoc [hunk 1]
```diff
@@ -35,16 +35,14 @@ $ oc adm catalog mirror \
...
+<5> Optional: Generate only the manifests required for mirroring, and do not actually mirror the image content to a registry. This option can be useful for reviewing what will be mirrored, and it allows you to make any changes to the mapping list if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images in a later step. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.
```
Comment (mjpytlak) [inline modules/olm-mirroring-catalog-colocated.adoc:45]:
```suggestion
<5> Optional: Generate only the manifests required for mirroring without actually mirroring the image content to a registry. This option can be useful for reviewing what will be mirrored and lets you make any changes to the mapping list, if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.
```

**Review by mjpytlak — COMMENTED** [review]
@xenolinux All of your changes LGTM. Thank you for being willing to make a few cop edits to improve the content.

File: modules/olm-mirroring-catalog-colocated.adoc [hunk 2]
```diff
@@ -35,16 +35,14 @@ $ oc adm catalog mirror \
     <mirror_registry>:<port>/<namespace> \ <2>
     [-a ${REG_CREDS}] \ <3>
     [--insecure] \ <4>
-    [--index-filter-by-os='<platform>/<arch>'] \ <5>
```
Comment (jianzhangbjz) [inline modules/olm-mirroring-catalog-colocated.adoc:38]: The `--index-filter-by-os='<platform>/<arch>'` still work, we should NOT remove it. But, the `--index-filter-by-os='.*'` doesn't work. We should remove the `.*` described in https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-installation-images.html#olm-mirror-catalog-colocated_installing-mirroring-installation-images, like the screenshot shows:
image

Comment (xenolinux) [reply modules/olm-mirroring-catalog-colocated.adoc:38]: Understood. Addressed. Thanks!

File: modules/olm-mirroring-catalog-colocated.adoc (hunk 1 above)
Comment (xenolinux) [reply modules/olm-mirroring-catalog-colocated.adoc]: Addressed

**Review by mjpytlak — COMMENTED** [review]
@xenolinux peer review LGTM.

File: modules/olm-mirroring-catalog-colocated.adoc [hunk 3]
```diff
@@ -43,8 +43,8 @@ $ oc adm catalog mirror \
...
+<5> Optional: Specify which platform and architecture of the index image to select when multiple variants are available. Images are passed as `'<platform>/<arch>[/<variant>]'`. This does not apply to images referenced by the index. Valid values are `linux/amd64`, `linux/ppc64le`, `linux/s390x`, `linux/arm64`.
+<6> Optional: Generate only the manifests required for mirroring without actually mirroring the image content to a registry. This option can be useful for reviewing what will be mirrored, and lets you to make any changes to the mapping list, if you require only a subset of packages. You can then use the `mapping.txt` file with the `oc image mirror` command to mirror the modified list of images in a later step. This flag is intended for only advanced selective mirroring of content from the catalog; the `opm index prune` command, if you used it previously to prune the index image, is suitable for most catalog management use cases.
```
Comment (lpettyjo) [inline modules/olm-mirroring-catalog-colocated.adoc:47]: Change: "and lets you to make any changes.." to "and lets you make any changes..." and allows you to make any changes..."

**Review by lpettyjo — COMMENTED** [review]
Otherwise, LGTM!

File: modules/olm-mirroring-catalog-colocated.adoc (hunk 3 above)
Comment (mjpytlak) [reply modules/olm-mirroring-catalog-colocated.adoc:47]: > Change: "and lets you to make any changes.." to "and lets you make any changes..." and allows you to make any changes..."

@lpettyjo This was part of an early copy edit. It should read `and lets you make any changes`. Good catch.

Comment (xenolinux) [reply modules/olm-mirroring-catalog-colocated.adoc]: @lpettyjo Addressed

**Review by bergerhoffer — COMMENTED** [review]
LGTM too, merging!

**Pull request merged**